src/app/code/community/Omise/Gateway/controllers/Callback/ValidateoffsiteinstallmentController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+class Omise_Gateway_Callback_ValidateoffsiteinstallmentController extends Omise_Gateway_Controller_Base
+{
+
+    public function indexAction()
+    {
+        // Callback validation.
+        $order = $this->_getOrder();
+
+        if (!$payment = $order->getPayment()) {
+            Mage::getSingleton('core/session')->addError(
+                $this->__('Installment payment validation failed, we cannot retrieve your payment information. Please contact our support team to confirm the payment.')
+            );
+
+            $this->_redirect('checkout/cart');
+            return;
+        }
+
+        $charge = Mage::getModel('omise_gateway/api_charge')->find(
+            $payment->getMethodInstance()->getInfoInstance()->getAdditionalInformation('omise_charge_id')
+        );
+
+        if ($charge instanceof Omise_Gateway_Model_Api_Error) {
+            Mage::getSingleton('core/session')->addError($charge->getMessage());
+            return $this->_redirect('checkout/cart');
+        }
+
+        if ($charge->isAwaitPayment()) {
+            $order->markAsAwaitPayment(
+                $payment->getLastTransId(),
+                Mage_Sales_Model_Order::STATE_PAYMENT_REVIEW,
+                Mage::helper('omise_gateway')->__('The payment has not been confirmed yet.<br/>It may take up to 24 hours to confirm your payment. In case of any questions please don’t hesitate to contact us.')
+            );
+            Mage::getSingleton('checkout/session')->addNotice(Mage::helper('omise_gateway')->__('Please note - the payment process is still ongoing. Once it is complete, you will receive the order confirmation.'));
+            return $this->_redirect('checkout/onepage/success');
+        }
+
+        if ($charge->isSuccessful()) {
+            $invoice = $order->getInvoice($payment->getLastTransId());
+
+            $order->markAsPaid(
```

Comment:
@jonrandy do we need to check for this edge case?
https://github.com/omise/omise-magento/pull/150/files